cli/internal/lockfile/pnpm_lockfile.go

```diff
@@ -255,30 +255,33 @@ func (p *PnpmLockfile) Patches() []turbopath.AnchoredUnixPath {
 	return patches
 }
 
-func (p *PnpmLockfile) resolveSpecifier(name string, specifier string) (string, bool) {
+func (p *PnpmLockfile) resolveSpecifier(workspace string, name string, specifier string) (string, bool) {
 	// Check if the specifier is already a resolved version
 	_, ok := p.Packages[formatPnpmKey(name, specifier)]
 	if ok {
 		return specifier, true
 	}
-	for workspacePkg, importer := range p.Importers {
-		for pkgName, pkgSpecifier := range importer.Specifiers {
-			if name == pkgName && specifier == pkgSpecifier {
-				if resolvedVersion, ok := importer.Dependencies[name]; ok {
-					return resolvedVersion, true
-				}
-				if resolvedVersion, ok := importer.DevDependencies[name]; ok {
-					return resolvedVersion, true
-				}
-				if resolvedVersion, ok := importer.OptionalDependencies[name]; ok {
-					return resolvedVersion, true
-				}
-
-				panic(fmt.Sprintf("Unable to find resolved version for %s@%s in %s", name, specifier, workspacePkg))
-			}
-		}
+	importer, ok := p.Importers[workspace]
+	if !ok {
+		return "", false
```

Comment:
I think we should probably panic in this scenario. If we're querying for dependencies of a non-existent workspace we're doing something wrong.

Reply:
Added a panic here, are there other comparisons in here where we should panic?